### PR TITLE
Fetch global data

### LINKED ISSUE #20: Fetch API Implimentation
- [x] Fetch User Data
- [x] Fetch Ingridient Data
- [x] Fetch Recipe Data
- [x] Impliment Dom manipulation with these data imports

### PR DESCRIPTION
**Pull request by stacyp2006**
###### What does this PR do?  
- Adds fetch methods to scripts.js to retrieve and implement recipe, ingredients and user data. 

###### Where should the reviewer start?  
- In the scripts.js, lines 8-34

###### How should this be manually tested?  
- On the browser, should still load with appropriate data. 

###### What are the relevant tickets?  
- close #20 

###### Questions:  
- Could this be refactored?

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -1,9 +1,38 @@
 // import './css/base.scss';
 import './css/styles.scss';
 
-import recipeData from './data/recipes';
```
Comment (nathanielmillard) [inline src/scripts.js:4]: Glad to see we are able to be completly functional without these lines of code.

File: src/scripts.js [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
-import users from './data/users';
+// import recipeData from './data/recipes';
+// import ingredientsData from './data/ingredients';
+// import users from './data/users';
+
+let users = [{}];
+let serverRecipeData = [{}];
+let ingredientsData = [{}];
+
+fetch('https://fe-apps.herokuapp.com/api/v1/whats-cookin/1911/users/wcUsersData')
```
Comment (nathanielmillard) [inline src/scripts.js:12]: To answers your question, I might look into if "nested" fetch requests exist, but currently this is looking pretty good to me.

File: src/user.js [hunk 3]
```diff
@@ -50,46 +50,51 @@ class User {
...
-          name: ingredient.name,
+          name: name,
           unit: ingredient.quantity.unit,
-          quantityNeeded: parseFloat(ingredient.quantity.amount - this.pantry.userIngredients[ingredient.name].toFixed(2))
+          quantityNeeded: parseFloat(ingredient.quantity.amount - this.pantry.userIngredients[ingredient.id].toFixed(2))
         }
       } else {
           neededIng = {
-            id: NaN,
+            id: ingredient.id,
```
Comment (nathanielmillard) [inline src/user.js:93]: Not ideal that we had to refactor this entire section to look at id's as opposed to names, but I think it was a pretty sussinct fix all things considered.

**Review by nathanielmillard — APPROVED** [review]
Over all looking good to me! 
I think this was a tackling of a new concept, and seems to work fine for asynconously executing these requested based on where they fall in the document.

File: src/scripts.js (hunk 2 above)
Comment (timkeresey) [reply src/scripts.js:12]: I figure nesting them would dictate the order in which they are fired.

File: src/user.js (hunk 3 above)
Comment (timkeresey) [reply src/user.js:93]: Yeah I think this works for what we need. We should come back to this one at the end and see how we can refactor it into smaller methods, though.

**Comment by timkeresey** [comment]
I like this implementation of fetch to get around the asynchronous of them. I'll definitely keep this in mind in the future.

**Pull request merged**